Phonebook.Backend/Phonebook.Backend.PictureService/Phonebook.Backend.PictureService/Startup.cs

```diff
@@ -0,0 +1,174 @@
+using KK.AspNetCore.Images.Processing;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Server.IISIntegration;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.FileProviders;
+using Phonebook.Backend.PictureService.Configuration;
+using Phonebook.Backend.PictureService.Controllers;
+using Swashbuckle.AspNetCore.Swagger;
+using System.ComponentModel.DataAnnotations;
+using Microsoft.AspNetCore.Mvc.ApiExplorer;
+using Microsoft.Extensions.Options;
+using Swashbuckle.AspNetCore.SwaggerGen;
+using Microsoft.Extensions.PlatformAbstractions;
+using System.Reflection;
+
+namespace Phonebook.Backend.PictureService
+{
+    /// <summary>
+    /// Startup Class
+    /// </summary>
+    public class Startup
+    {
+        private IConfiguration Configuration { get; }
+        private PictureServiceConfiguration AppSettings { get; }
+        private const string CorsPolicy = "AllowDomainList";
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Startup"/> class.
+        /// </summary>
+        /// <param name="configuration"></param>
+        public Startup(IConfiguration configuration)
+        {
+            this.Configuration = configuration;
+            this.AppSettings = configuration.Get<PictureServiceConfiguration>();
+        }
+
+        // This method gets called by the runtime. Use this method to add services to the container.
+        public void ConfigureServices(IServiceCollection services)
+        {
+            // Validation of the appsettings on Startup
+            services.AddSingleton<PictureServiceConfiguration>(AppSettings);
+            services.UseConfigurationValidation();
+            services.ConfigureValidatableSetting<PictureServiceConfiguration>(Configuration);
+            services.ConfigureValidatableSetting<ContactInformation>(Configuration.GetSection("ContactInformation"));
+            services.AddSingleton<PictureServiceConfiguration>(AppSettings);
+
+            // Add Api Versioning Logic
+            services.AddApiVersioning(o =>
+            {
+                o.ReportApiVersions = true;
+                o.AssumeDefaultVersionWhenUnspecified = true;
+                o.DefaultApiVersion = new ApiVersion(1, 0);
+
+            });
+
+            services.AddVersionedApiExplorer(options =>
+            {
+                // add the versioned api explorer, which also adds IApiVersionDescriptionProvider service
+                // note: the specified format code will format the version as "'v'major[.minor][-status]"
+                options.GroupNameFormat = "'v'VVV";
+                // note: this option is only necessary when versioning by url segment. the SubstitutionFormat
+                // can also be used to control the format of the API version in route templates
+                options.SubstituteApiVersionInUrl = true;

```

Comment:
Please do not version in the URL. We can have a debate if you want to :)

Reply:
I would prefer query string or Header based versioning

Reply:
thats the configuration for the Api Explorer. Not for the normal api endpoints. It would very helpful if we can create an "api" route path to make the differenz more clear. Because if you call:

`users/whoami`

you can do `?api-version=2.0`. 

But if you do the same on `swagger` it will show version 1. Because swagger works with another behavior without make it clear by the path. And this setting isn't very good documented. It does not a difference at all if you change the value from true to false. So maybe we must create an documentation issue in the apiExplorer project. 

So currently there is no plan to change this line, if we doesn't know what is does.